charts/ingress-nginx/values.yaml

```diff
@@ -18,7 +18,17 @@ ingress-nginx:
                   operator: In
                   values:
                     - ingress-nginx
-            topologyKey: "topology.cast.ai/csp"
+            topologyKey: "kubernetes.io/hostname"
+        preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 100
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                  - key: app.kubernetes.io/name
+                    operator: In
+                    values:
+                      - ingress-nginx
+              topologyKey: "topology.cast.ai/csp"
     metrics:
```

Comment:
Maybe adding podAntiAffinity to make sure nginx replicas won't sit on a single node?

Reply:
Yes, it's already added, see requiredDuringSchedulingIgnoredDuringExecution rule.